privacyidea/static/components/login/controllers/loginControllers.js

```diff
@@ -322,9 +322,9 @@ angular.module("privacyideaApp")
                 }
             }
         }).finally(function () {
-            // We delete the login object, so that the password is not
+            // We delete the password from the login object, so that it is not
             // contained in the scope
-            $scope.login = {username: "", password: ""};
+            $scope.login.password = "";
```

Comment:
Please check, if we can do:

~~~~javascript
if ($scope.transactionid) {
   $scope.login.password = "";
} else {
   $scope.login = {username: "", password: ""};
}
~~~~

This way we would drop the username for invalid auth and keep the username only in case we do a challenge response.

Reply:
The old behavior was, that only if the authentication succeeded, the login-data was removed.
If the auth request failed (wrong pw or challenge-response) the password was removed immediately (and if it wasn't challenge-response the transaction id was removed later) and only the username was kept.
So my idea is, we keep the error handler as is and just remove the login data in the success-handler.

Reply:
Done in f7671b66
Tested with several chal-resp tokens (U2F, WebAuthN, Email, HOTP).